Mathlib/Combinatorics/Additive/PluenneckeRuzsa.lean

```diff
@@ -50,31 +56,67 @@ theorem ruzsa_triangle_inequality_div_div_div (A B C : Finset α) :
     · exact div_right_injective (Prod.ext_iff.1 h).1
   · exact ((mem_bipartiteBelow _).1 hv).2
 
+/-- **Ruzsa's triangle inequality**. Invmul-invmul-invmul version. -/
+@[to_additive "**Ruzsa's triangle inequality**. Negadd-negadd-negadd version."]
+theorem ruzsa_triangle_inequality_inv_mul_inv_mul_inv_mul (A B C : Finset G) :
+    #(C⁻¹ * A) * #B ≤ #(B⁻¹ * A) * #(C⁻¹ * B) := by
+  simpa [div_eq_mul_inv, ← map_op_mul, ← map_op_inv] using
+    ruzsa_triangle_inequality_div_div_div (G := Gᵐᵒᵖ) (A.map opEquiv.toEmbedding)
+      (B.map opEquiv.toEmbedding) (C.map opEquiv.toEmbedding)
+
 /-- **Ruzsa's triangle inequality**. Div-mul-mul version. -/
 @[to_additive "**Ruzsa's triangle inequality**. Sub-add-add version."]
-theorem ruzsa_triangle_inequality_div_mul_mul (A B C : Finset α) :
-    (A / C).card * B.card ≤ (A * B).card * (B * C).card := by
-  rw [← div_inv_eq_mul, ← card_inv B, ← card_inv (B * C), mul_inv, ← div_eq_mul_inv]
+theorem ruzsa_triangle_inequality_div_mul_mul (A B C : Finset G) :
+    #(A / C) * #B ≤ #(A * B) * #(C * B) := by
+  rw [← div_inv_eq_mul, ← card_inv B, ← card_inv (C * B), mul_inv_rev, ← div_eq_mul_inv]
   exact ruzsa_triangle_inequality_div_div_div _ _ _
 
+/-- **Ruzsa's triangle inequality**. Invmul-mul-mul version. -/
+@[to_additive "**Ruzsa's triangle inequality**. Negadd-add-add version."]
+theorem ruzsa_triangle_inequality_inv_mul_mul_mul (A B C : Finset G) :
+    #(C⁻¹ * A) * #B ≤ #(B * A) * #(B * C) := by
+  simpa [div_eq_mul_inv, ← map_op_mul, ← map_op_inv] using
+    ruzsa_triangle_inequality_div_mul_mul (G := Gᵐᵒᵖ) (A.map opEquiv.toEmbedding)
+      (B.map opEquiv.toEmbedding) (C.map opEquiv.toEmbedding)
+
 /-- **Ruzsa's triangle inequality**. Mul-div-mul version. -/
 @[to_additive "**Ruzsa's triangle inequality**. Add-sub-add version."]
-theorem ruzsa_triangle_inequality_mul_div_mul (A B C : Finset α) :
-    (A * C).card * B.card ≤ (A / B).card * (B * C).card := by
+theorem ruzsa_triangle_inequality_mul_div_mul (A B C : Finset G) :
+    #(A * C) * #B ≤ #(A / B) * #(B * C) := by
```

Comment:
What side should the `B` be on here then?

Reply:
This case is ambiguous, so you can put it where you like